https://bugs.gentoo.org/919399
https://github.com/nhorman/rng-tools/pull/205

From 593e326815ba7ee820ee2f0a3c45279282f07c34 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Tue, 19 Dec 2023 10:32:11 +0100
Subject: [PATCH] configure: Fix m4 quoting issue in libargp probe

Without the surrounding [], m4 drops the inner [], treating, argv[] as
argv.  This results in an incorrect argument type for argp_parse,
causing the probe to fail with future compilers due to type error.
--- a/configure.ac
+++ b/configure.ac
@@ -196,7 +196,7 @@ AS_IF(
 		AC_LINK_IFELSE(
 			[AC_LANG_PROGRAM(
 				[#include <argp.h>],
-				[int argc=1; char *argv[]={"test"}; argp_parse(0,argc,argv,0,0,0); return 0;]
+				[[int argc=1; char *argv[]={"test"}; argp_parse(0,argc,argv,0,0,0); return 0;]]
 				)],
 			[need_libargp=no],
 			[need_libargp=yes
